examples/image-class.lisp

```diff
@@ -0,0 +1,73 @@
+(defpackage image
+  (:use #:cl
+        #:alexandria)
+  (:import-from #:serapeum
+                #:lret
+                #:in))
+(in-package #:image)
+
+(defclass image ()
+  ((wand :initarg :wand
+         :accessor image-wand
+         :initform (magick:new-magick-wand))))
+
+
+(defmethod initialize-instance :after ((img image) &key)
+  (trivial-garbage:finalize img
+                            (lambda () (magick:destroy-pixel-wand (image-wand img)))))
```

Comment:
Aren't you closing over the image here, preventing it from being finalized? You can bind the value of image-wand and close over that.

Reply:
I looked at examples of  `trivial-garbage` and it looks like it will be finalized anyway:
https://github.com/marcoheisig/Petalisp/blob/master/code/native-backend/backend.lisp
https://github.com/hahahahaman/turtle-geometry/blob/master/src/drawer/drawer.lisp

Reply:
If you look at the first example, you can see they're introducing a binding for just the worker pool so the finalizer isn't closing over the whole backend (which is what I was talking about here). 

I think the second example is just bugged: they should be introducing a local binding for vao instead of just using with-slots.

Reply:
The SBCL manual discusses what I'm talking about: https://www.sbcl.org/manual/#index-finalize

Reply:
I've opened an issue on the turtle-graphics repository: https://github.com/hahahahaman/turtle-geometry/issues/1